src/components/MenuButtons.vue

```diff
@@ -0,0 +1,43 @@
+<!-- Requires the icon filename within src/assets/UI/menuBtnIcons as a png -->
+<!-- Every button should open up a modal when clicked, pass callbacks for specific modal handlers -->
+<script setup lang="ts">
+	const props = defineProps<{
+		btnInfoArray: {
+			imgFileName: string;
+			tooltip: string;
+			modalHandler: () => void;
+		}[];
+	}>();
+</script>
+
+<template>
+	<div class="menuBtnContainer">
+		<button
+			v-for="(btn, btnIndex) in props.btnInfoArray"
+			:key="`btn-${btnIndex}`"
+			class="menuBtn"
+			:style="{ backgroundImage: `url('src/assets/UI/menuBtnIcons/${btn.imgFileName}.png')` }"
+			:title="btn.tooltip"
+			@click.prevent="btn.modalHandler"></button>
+	</div>
+</template>
+
+<style scoped>
+	/* Based on current design, buttons are in a 2x2 grid, can change grid-template-columns for adding/removing buttons */
+	.menuBtnContainer {
+		display: inline-grid;
+		grid-template-columns: 1fr 1fr;
```

Comment:
Use v-bind to dynamically create an appropriate amount of columns - probably with flex.

Reply:
Not using v-bind here. There was a simple css solution. using calc(),